### DISCUSSION TITLE
What is the best way to run the code from nvchad?

### DISCUSSION THREAD

**Discussion by DAT4** [NvChad/NvChad#2273] [Q&A]
How can I create mappings for running or compiling code in different programming languages in NeoVim, considering the availability of a pop-up terminal and the need to run the entire project or specific examples/tests for larger projects in Go and Rust? Are there any existing plugins that could help with this?

Explanation:

I have been successfully using Vim with autocmd's to map the [Enter] key to run or compile the code I am working on for various programming languages. However, as I switch to NeoVim (NVChad), I want to adapt this functionality to the new environment. In NVChad, I have the option of using a pop-up terminal, so I need to consider how I want to display the output when running the code. Additionally, I now work on larger projects in Go and Rust, so I'm looking for a way to run the entire project or specific examples/tests easily.

I believe having a plugin that allows me to run makefiles from within NeoVim and select the desired make target (e.g., make run, make example-1, etc.) would be useful for my workflow.

I am curious to know if there is a standard or universal approach for achieving this, or if there are any existing plugins that can help with my specific requirements.

Thank you for your assistance.

**Comment by moisesmorillo** [comment]
This plugin can help you [https://github.com/stevearc/overseer.nvim](https://github.com/stevearc/overseer.nvim) it recognizes Makefiles, and you can also create tasks (like vscode) in your repos and run them when need it.
